src/main/java/bio/terra/service/tabulardata/google/BigQueryProjectFactory.java

```diff
@@ -0,0 +1,61 @@
+package bio.terra.service.tabulardata.google;
+
+import bio.terra.service.filedata.google.gcs.GcsConfiguration;
+import com.google.cloud.bigquery.BigQuery;
+import com.google.cloud.bigquery.BigQueryOptions;
+import com.google.cloud.http.HttpTransportOptions;
+import com.google.cloud.storage.StorageOptions;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.context.properties.ConfigurationProperties;
+import org.springframework.context.annotation.Configuration;
+import org.springframework.context.annotation.Profile;
+import org.springframework.stereotype.Component;
+
+@Configuration
+@Profile("google")
+@ConfigurationProperties(prefix = "datarepo.gcs.bq")
+public class BigQueryProjectFactory {
+  private static final Logger logger = LoggerFactory.getLogger(BigQueryProjectFactory.class);
+  private int connectTimeoutSeconds;
+  private int readTimeoutSeconds;
+
+  @Autowired
+  public BigQueryProjectFactory() {
+    // this.gcsConfiguration = gcsConfiguration;
+  }
+
+  public static BigQueryProject buildBQProject(String projectId) {
+    logger.info("Retrieving Bigquery project for project id: {}", projectId);
```

Comment:
This is close.  You want this to be a non-static method then you can use the `gcsConfiguration` member variable.  The thing would then be to use this factory in your calling classes (where it can be autowired in)

Reply:
I'm not following - If I could autowire something in BigQueryProject (the calling class), then couldn't I just autowire gcsConfiguration?

Reply:
I was thinking you could autowire in the factory class and use that to create BQprojects

Reply:
It still requires putting the factory in all of the calling classes though

Reply:
Many classes already inject `ApplicationConfiguration`, so it might be better to wire these settings into that config instead of creating a new config class for them.